### ISSUE TITLE
[Bug]: HELM + KAMEL installation with insecure internal registry fails while pulling the kit

### ISSUE THREAD

**Issue by RaffaelGrob** [apache/camel-k#4720]
### What happened?

I have the operator installed to use an internal registry listening on a service within the same namespace (**HTTP**://my-registry.my-namespace.svc.cluster.local:80). No TLS is configured. When the Integration should create a Pod, Kubernetes tries to pull the image by **HTTPS** (rather then HTTP) and fails. https://my-registry.my-namespace.svc.cluster.local:80/v2/my-nam.....

My expectation is,
a. Exactly the same registry url is used to push and to pull the image
b. that when i use the "insecure: true" configuration, the image pull uses HTTP

### Steps to reproduce

Reproduction:
1. Prepared the operators by helmchart. Here the relevant part of the values.yaml
   ```yaml
   platform:
    build:
      registry:
        address: my-registry.my-namespace.svc.cluster.local:80
        insecure: true
   ```
2. `kamel uninstall` to cleanup
3. `kamel reset all` to cleanup
4.  deploy helmchart
	- check operator logs for errors: *nothing suspicious*
	- check registry to see if `my-namespace/camel-k-runtime-quarkus-builder` image exists: *success*
5. deploy my integration
	- observe the operator: *pushing to registry successful*
		- Observed this new image `my-namespace/camel-k-kit-cjs4g8f1jepq5b30qp20`: *success*
6. ERROR: Happens when the integration tries to pull the image that was previously pushed successfully. 
   Warning  Failed     41m (x4 over 43m)      kubelet            Error: ErrImagePull
   Warning  Failed     41m (x6 over 43m)      kubelet            Error: ImagePullBackOff
   Normal   BackOff    2m58s (x174 over 43m)  kubelet            Back-off pulling image

### Relevant log output

```shell
kubectl logs camel-k-operator-896747ccb-zpbbv --since 12h | jq '. | select(.level != "info")'
-> No Errors found

kubectl describe pod welcome-route2-5c9bd4d976-6mr4d

Name:             welcome-route2-5c9bd4d976-6mr4d
...
Labels:           camel.apache.org/integration=welcome-route2
                  pod-template-hash=5c9bd4d976
Annotations:      camel.apache.org/operator.id: camel-k-dev
...
Controlled By:  ReplicaSet/welcome-route2-5c9bd4d976
Containers:
  integration:
    Container ID:    
    Image:           my-registry.my-namespace.svc.cluster.local:80/my-namespace/camel-k-kit-cjs4g8f1jepq5b30qp20@sha256:008cb6a4505cd74ec2d7bbee5d3efde6c002f3d6bd5559fb0bb13a67b17721d9
    Image ID:        
  
Events:
  Type     Reason     Age                    From               Message
  ----     ------     ----                   ----               -------
  Normal   Scheduled  43m                    default-scheduler  Successfully assigned my-namespace/welcome-route2-5c9bd4d976-6mr4d to ip-10-130-118-220.eu-central-1.compute.internal
  Normal   Pulling    41m (x4 over 43m)      kubelet            Pulling image "my-registry.my-namespace.svc.cluster.local:80/my-namespace/camel-k-kit-cjs4g8f1jepq5b30qp20@sha256:008cb6a4505cd74ec2d7bbee5d3efde6c002f3d6bd5559fb0bb13a67b17721d9"
  Warning  Failed     41m (x4 over 43m)      kubelet            Failed to pull image "my-registry.my-namespace.svc.cluster.local:80/my-namespace/camel-k-kit-cjs4g8f1jepq5b30qp20@sha256:008cb6a4505cd74ec2d7bbee5d3efde6c002f3d6bd5559fb0bb13a67b17721d9": rpc error: code = Unknown desc = failed to pull and unpack image "my-registry.my-namespace.svc.cluster.local:80/my-namespace/camel-k-kit-cjs4g8f1jepq5b30qp20@sha256:008cb6a4505cd74ec2d7bbee5d3efde6c002f3d6bd5559fb0bb13a67b17721d9": failed to resolve reference "my-registry.my-namespace.svc.cluster.local:80/my-namespace/camel-k-kit-cjs4g8f1jepq5b30qp20@sha256:008cb6a4505cd74ec2d7bbee5d3efde6c002f3d6bd5559fb0bb13a67b17721d9": failed to do request: Head "https://my-registry.my-namespace.svc.cluster.local:80/v2/my-namespace/camel-k-kit-cjs4g8f1jepq5b30qp20/manifests/sha256:008cb6a4505cd74ec2d7bbee5d3efde6c002f3d6bd5559fb0bb13a67b17721d9": dial tcp: lookup my-registry.my-namespace.svc.cluster.local: no such host
  Warning  Failed     41m (x4 over 43m)      kubelet            Error: ErrImagePull
  Warning  Failed     41m (x6 over 43m)      kubelet            Error: ImagePullBackOff
  Normal   BackOff    2m58s (x174 over 43m)  kubelet            Back-off pulling image "my-registry.my-namespace.svc.cluster.local:80/my-namespace/camel-k-kit-cjs4g8f1jepq5b30qp20@sha256:008cb6a4505cd74ec2d7bbee5d3efde6c002f3d6bd5559fb0bb13a67b17721d9"
```

### Camel K version

2.0.0

**Comment by squakez** [comment]
Yes, this is some known problem on Kubernetes, see the discussion thread [1]. If you want to use an insecure registry, it seems that the only way is to configure your Kubernetes cluster to be allowed to do so [2]. I guess that it would be better to make an effort and try to configure a secure registry and avoid all those workarounds though.

[1] https://camel.zulipchat.com/#narrow/stream/257299-camel-k/topic/Registry.20with.20no.20credentials
[2] https://stackoverflow.com/questions/51893120/kubernetes-pull-from-insecure-docker-registry

**Comment by RaffaelGrob** [comment]
Thank you for having a look at it. My conclusion after reading the links above is, that we should use a secure registry instead. In parallel with the attempt described in this Issue, I did setup a secure private registry with a internal certificate. But this didn't work either and I created this issue for it: [issues/4724](https://github.com/apache/camel-k/issues/4724)

The problem I describe here comes from the fact, that the AWS ECR seems incompatible with our Camel-K setup ("doesn't support folders", "the image name is the tag") and therefore we're looking for a Registry alternative.

**Comment by squakez** [comment]
> The problem I describe here comes from the fact, that the AWS ECR seems incompatible with our Camel-K setup ("doesn't support folders", "the image name is the tag") and therefore we're looking for a Registry alternative.

I guess this is a limitation of the registry then. As discussed in several occasions, we cannot support all the different registry solutions, so we try to stick to the standards (in this case the one proposed in the [docker tag](https://docs.docker.com/engine/reference/commandline/tag/)).

**Comment by squakez** [comment]
We should include the last statement somewhere in the registry documentation.
